### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by SageV-add123**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,28 +20,37 @@
...
+    "@babel/preset-env": "^7.22.20",
     "@babel/preset-react": "^7.10.1",
+    "@babel/preset-typescript": "^7.23.0",
     "@babel/runtime": "^7.10.2",
     "jest": "^24.9.0",
-    "standard": "^14.3.4"
+    "ts-standard": "^12.0.2"
   },
   "peerDependencies": {
+    "amiable-forms": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (SVadd123) [reply package.json:33]: This _is_ a version bump. It's no longer a "release candidate".

File: package.json [hunk 2]
```diff
@@ -1,13 +1,14 @@
 {
   "name": "add123-manifest",
-  "version": "2.1.0",
+  "version": "2.2.0",
```
[the remark this answers is not in the document]
Comment (SVadd123) [reply package.json:3]: Ideally this is a strict enhancement, and should not involve a change in the API for consumers of the library.

File: src/components/Filter.tsx [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
   const { updateState: updateManifestState } = useManifest()
   const { setValues: updateForm } = useForm({ shouldUpdate: NOPE })
-  const filterRef = useRef()
+  const filterRef = useRef<UpdaterFilter>()
 
   useEffect(() => {
     const updatedFilter = {
-      ...(defaultValues || {}),
-      ...(defaultTableState || {}),
+      ...(defaultValues ?? {}),
```
Comment (calebsoper) [inline src/components/Filter.tsx:33]: good change

Comment (SVadd123) [reply src/components/Filter.tsx:33]: Plus it was this or have typescript point a gun (shaped a lot like the word 'any') at me

**Review by calebsoper — APPROVED** [review]
TS changes look good from where I'm standing!

**Review by NikB0t — APPROVED** [review]

**Pull request merged**